OMEdit/OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp

```diff
@@ -7262,6 +7262,38 @@ void ModelWidgetContainer::currentModelWidgetChanged(QMdiSubWindow *pSubWindow)
     }
     // update the Undo/Redo actions
     pModelWidget->updateUndoRedoActions();
+    /* ticket:5441 OMEdit toolbars
+     * Show the relevant toolbars if we are in a Modeling perspective
+     */
+    if (MainWindow::instance()->getPerspectiveTabBar()->currentIndex() == 1) {
+      if (pModelWidget->getLibraryTreeItem()->getLibraryType() == LibraryTreeItem::Modelica) {
+        MainWindow::instance()->getShapesToolBar()->setVisible(true);
+        MainWindow::instance()->getCheckToolBar()->setVisible(true);
+        MainWindow::instance()->getSimulationToolBar()->setVisible(true);
+        MainWindow::instance()->getTLMSimulationToolbar()->setVisible(false);
+        MainWindow::instance()->getOMSimulatorToobar()->setVisible(false);
+      } else if (pModelWidget->getLibraryTreeItem()->getLibraryType() == LibraryTreeItem::Text) {
```

Comment:
Maybe have a function that returns true if LibraryTreeItem::Text || LibraryTreeItem::CompositeModel || LibraryTreeItem::OMS is true then we would only need an if elseif and an else I think

Reply:
I will pay you a visit and explain myself 💯